torch/_refs/__init__.py

```diff
@@ -2055,6 +2063,14 @@ def tensor_split(
         return tuple(splits)
 
 
+def dsplit(a: TensorLikeType, sections: DimsType) -> TensorSequenceType:
+    if a.ndim < 3:
+        raise RuntimeError(f"torch.dsplit requires a tensor with at least 3 dimension, but got a tensor with {a.ndim} dimensions!")
+    if isinstance(sections, int) and (sections == 0 or a.shape[2] % sections != 0):
+        raise RuntimeError(f"torch._refs.dsplit attempted to split along dimension 2, but the size of the dimension {a.shape[2]} is not divisible by the split_size {sections}!")
```

Comment:
Is this error check redundant with calling tensor_split?

Reply:
it is actually not.

`a.shape[2] % sections != 0` this should only apply to dsplit. (double checked with numpy as well as torch)
`sections == 0` <- this one will actually also error out on tensor_split, but error message is not consistent with `torch.dsplit`